packages/manifold/contracts/lazyclaim/LazyPayableClaim.sol

```diff
@@ -208,6 +216,17 @@ abstract contract LazyPayableClaim is ILazyPayableClaim, AdminControl {
         _claimMintIndices[creatorContractAddress][instanceId][claimMintIndex] = claimMintTracking | mintBitmask;
     }
 
+    function _checkSignatureAndUpdate(address creatorContractAddress, uint256 instanceId, bytes calldata signature, bytes32 message, bytes32 nonce, address signingAddress, address mintFor, uint256 expiration) private {
+        // Verify valid message based on input variables
+        bytes32 expectedMessage = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", instanceId, nonce, mintFor, expiration));
+        // Verify nonce usage/re-use
+        require(!_usedMessages[creatorContractAddress][instanceId][expectedMessage], "Cannot replay transaction");
```

Comment:
This is WRONG!

You need to check the nonce, otherwise I can replay with the same nonce with different mintFor addresses.

Reply:
Add regression test for this.

Reply:
Good catch. Or same `mintFor` and different `expiration`.